client/src/standalone/videos/embed-api.ts

```diff
@@ -195,4 +197,18 @@ export class PeerTubeEmbedApi {
   private isWebVideo () {
     return !!this.embed.player.webVideo
   }
+
+  private getImageDataUrl (): string {
+
+    const video = this.element
+
+    const canvas = document.createElement('canvas')
+
+    canvas.width = video.videoWidth
+    canvas.height = video.videoHeight
+
+    canvas.getContext('2d').drawImage(video, 0, 0, canvas.width, canvas.height)
+
+    return canvas.toDataURL('image/jpeg')
```

Comment:
Do you think we can generate a blob directly so we don't have to change server code? https://stackoverflow.com/questions/6850276/how-to-convert-dataurl-to-file-object-in-javascript

Reply:
I can create as a blob, but I'm unable to figure out how to attach it such that the server code will detect it as an attached file.

Reply:
I was able to get this one figured out. Just had to create a File object with the proper MIME type to set as the form value.